src/main/java/com/bobocode/bibernate/QueryHelper.java

```diff
@@ -0,0 +1,30 @@
+package com.bobocode.bibernate;
+
+import com.bobocode.bibernate.exception.QueryHelperException;
+import com.bobocode.bibernate.session.Session;
+import com.bobocode.bibernate.session.SessionFactory;
+
+import java.util.function.Function;
+
+public class QueryHelper {
+    private SessionFactory sessionFactory;
+
+    public QueryHelper(SessionFactory sessionFactoryFactory) {
+        this.sessionFactory = sessionFactoryFactory;
+    }
+
+    public <T> T readWithinTx(Function<Session, T> entityManagerConsumer) {
+        Session session = sessionFactory.createSession();
+        session.beginTransaction();
+        try {
+            T result = entityManagerConsumer.apply(session);
+            session.commitTransaction();
+            return result;
+        } catch (Exception e) {
+            session.rollbackTransaction();
+            throw new QueryHelperException("Transaction is rolled back", e);
```

Comment:
In my opinion, we don't need a new custom exception for this class, we can just reuse the generic BibernateException here. What do you think?